Add 'xray_evidence' fixture

ronanviel opened the pull request:
Hello,

Here is a new proposal for a another way to add an evidence to a Test Run. My motivations were:

1. to avoid update of conftest.py by developers. In large project, it can create errors and conflicts,
2. to add specific evidence to a test without writing conditions in conftest.py,
3. to keep everything related to a test in the same file which is easier to develop and to maintain.

The propose solution is a new fixture, named 'xray_evidence'.

It takes a path to a file to upload or to name 'data' passed as argument. If filename extension is unknown of the fixture, it defaults to 'application/octet-stream', unless the 'ctype' argument provides a content-type. Evidence file created in the current working directory (of pytest execution) is included even if the test source file is in a subdirectory.

Tests are included to cover 100% of the added code.

Other minor changes are proposed for code I wrote and you already merged:  

- Fix typo in content type 'text/plain' (was 'plain/text)
- Add option "--add-captures" documentation (was missing)
- Improve coverage by testing --add-captures with error message in report.longrepr
- Update to conform to "new code should avoid using testdir in favor of pytester"

fundakol commented on src/pytest_xray/evidence.py:
```diff
@@ -50,3 +53,72 @@ def json(data: AnyStr, filename: str) -> Dict[str, str]:
+                         *, data: Union[str, bytes] = '',
+                         ctype: str = ''
+                         ) -> None:
+        if (not isinstance(path, (str, Path)) or path == ''):
```
Parentheses are redundant.

fundakol commented on src/pytest_xray/helper.py:
```diff
@@ -100,6 +100,9 @@ def merge(self, other: 'TestCase') -> None:
                 self.comment += ('\n' + '-'*80 + '\n')
                 self.comment += other.comment
 
+        if other.evidences != []:
```
Can be simpler: 
```
if other.evidences:
```

fundakol commented on src/pytest_xray/evidence.py:
```diff
@@ -50,3 +53,72 @@ def json(data: AnyStr, filename: str) -> Dict[str, str]:
+
+
+@pytest.fixture
+def xray_evidence(request) -> Callable:
```
Can you add this fixture to `XrayPlugin` class, and remove import `from pytest_xray.evidence import xray_evidence # noqa: F401` from `plugin.py`?
I would like to keep whole business logic in the plugin implementation.

fundakol commented on src/pytest_xray/evidence.py:
```diff
@@ -50,3 +53,72 @@ def json(data: AnyStr, filename: str) -> Dict[str, str]:
+def xray_evidence(request) -> Callable:
+    """ Fixture to add an evidence to the Test Run details of a Test.
+
+    Copyright © 2023 Orange - All rights reserved
```
This repo is under Apache license. I am not an expert for licenses, that why I would like to avoid having some copyrights that may not be compatible with Apache license.

fundakol commented on src/pytest_xray/evidence.py:
```diff
@@ -50,3 +53,72 @@ def json(data: AnyStr, filename: str) -> Dict[str, str]:
+        if ctype == '':
+            extension = evidence_path.suffix[1:]
+            if media_types.get(extension) is None:
+                contentType = 'application/octet-stream'
```
Can you use camel case for the variable names?

fundakol reviewed (changes requested):
Thanks for the PR. I looks good, but I had few minor comments.

ronanviel commented:
I have updated the PR as you required. I hope you will find it good for merging.
I have removed the copyright from the source after checking with my legal.

fundakol commented:
Can you amend your commit and put one empty line between the commit header description and detailed description. This helps to print log with `--oneline` option:
```
git log --oneline
```

Also I found a bug in my code which was fixed by your change :thumbsup:, but also your change introduces new bug. I added two tests that covers these cases: https://github.com/fundakol/pytest-jira-xray/blob/master/tests/test_evidence_with_duplicated_ids.py

ronanviel commented:
Thank you for your feedback. I am going to work on this next week.
